doc/source/building/index.rst

```diff
@@ -327,24 +327,24 @@ virtual environments:
 
         ::
 
-          python -m venv venv
-          source venv/bin/activate
+          python -m venv .venv
+          source .venv/bin/activate
```

Comment:
If we're going to do this, I suspect we should also change the line above that reads:

> Create and activate a virtual environment in a new directory named venv 

I'm hesitant to start more bikeshedding, but can't resist wondering why we're recommending an in-source `venv` instead of something more durable/less awkward. Is it really considered a desirable feature to lose your `venv` when you do i.e., `git clean -xfd`?

Reply:
> If we're going to do this, I suspect we should also change the line above that reads:
> 
> > Create and activate a virtual environment in a new directory named venv

That's done.

As an alternative, `.gitignore` can be changed to mention `venv` instead of `.venv`.

I'll have to defer the question about whether the environment should be in the source directory back to you guys.

Reply:
> but can't resist wondering why we're recommending an in-source `venv` instead of something more durable/less awkward. Is it really considered a desirable feature to lose your `venv` when you do i.e., `git clean -xfd`?

I think the common perception among scientific Python maintainers is that this is indeed bad practice and has more downsides than upsides. But it is increasingly popular among folks working on lightweight pure Python packages and it is advocated for by PyPA folks. So users/contributors are going to be doing this. I think we (a) have to make sure it works well in CI (which we do), and (b) we use it in the docs here because it's easier than dealing with paths outside the repo. 

I'd treat it like a bikeshed at this point indeed, and preferably not touch it.